stopify/src/runtime/abstractRunner.ts

```diff
@@ -64,8 +83,37 @@ export abstract class AbstractRunner implements AsyncRun {
     const estimator = makeEstimator(this.opts);
     this.suspendRTS = new RuntimeWithSuspend(this.continuationsRTS,
       this.opts.yieldInterval, estimator);
-    this.suspendRTS.mayYield = () => this.mayYieldRunning();
-    this.suspendRTS.onYield = () => this.onYieldRunning();
+    this.suspendRTS.mayYield = () => {
+      if(this.mayYieldFlag === mayYieldFlagValues.Resume) {
+       return this.mayYieldRunning();
+      } else { //Step
+        // Yield control if the line number changes.
+        const maybeLine = this.suspendRTS.linenum;
+        if (typeof maybeLine !== 'number' || maybeLine === this.captureCurrentLine) {
+          return false;
+        } else {
+          this.captureOnStepFn(maybeLine);
+          return true;
+        }
+      }
+    };
+    this.suspendRTS.onYield = () => {
+      if(this.onYieldFlag === onYieldFlagValues.Paused) {
+        this.onYieldFlag = onYieldFlagValues.PausedAndMayYield;
+        const maybeLine = this.suspendRTS.linenum;
+        typeof maybeLine === 'number' ? this.capturePauseFn(maybeLine) : this.capturePauseFn();
```

Comment:
You can just call `this.capturePauseFn(maybeLine)`, no need for the ternary or `typeof` check.